MatrixSDK/Background/MXBackgroundStore.swift

```diff
@@ -83,45 +58,177 @@ class MXBackgroundStore: MXMemoryStore {
         get {
             return fileStore.syncFilterId
         } set {
             //  no-op
         }
     }
     
+    func event(withEventId eventId: String, inRoom roomId: String) -> MXEvent? {
+        guard let roomStore = roomStore(forRoom: roomId) else {
+            return nil
+        }
+        
+        let event = roomStore.event(withEventId: eventId)
+        
+        NSLog("[MXBackgroundStore] eventWithEventId: \(eventId) \(event == nil ? "not " : "" )found")
+        return event
+    }
+    
+    
+    //  MARK: - Private
+    private func roomStore(forRoom roomId: String) -> MXFileRoomStore? {
+        // Use the cached instance if available
+        if let roomStore = roomsStore[roomId] {
+            return roomStore
+        }
+        
+        guard let roomStore = fileStore.roomStore(forRoom: roomId) else {
+            NSLog("[MXBackgroundStore] roomStore: Unknown room id: \(roomId)")
+            return nil
+        }
+        
+        roomsStore[roomId] = roomStore
+        return roomStore
+    }
+    
+    
+    //  MARK: - Stubs
+    
+    /// Following operations should be not required
+    
+    func open(with credentials: MXCredentials, onComplete: (() -> Void)?, failure: ((Error?) -> Void)? = nil) {
+    }
+    
+    func storeEvent(forRoom roomId: String, event: MXEvent, direction: __MXTimelineDirection) {
+    }
+    
+    func replace(_ event: MXEvent, inRoom roomId: String) {
+    }
+    
+    func eventExists(withEventId eventId: String, inRoom roomId: String) -> Bool {
+        return false
+    }
+    
+    func deleteAllMessages(inRoom roomId: String) {
+    }
+    
+    func deleteRoom(_ roomId: String) {
+    }
+    
+    func deleteAllData() {
+    }
+    
+    func storePaginationToken(ofRoom roomId: String, andToken token: String) {
+    }
+    
+    func paginationToken(ofRoom roomId: String) -> String? {
+        return nil
+    }
+    
+    func storeHasReachedHomeServerPaginationEnd(forRoom roomId: String, andValue value: Bool) {
+    }
+    
+    func hasReachedHomeServerPaginationEnd(forRoom roomId: String) -> Bool {
+        return true
+    }
+    
+    func storeHasLoadedAllRoomMembers(forRoom roomId: String, andValue value: Bool) {
+    }
+    
+    func hasLoadedAllRoomMembers(forRoom roomId: String) -> Bool {
+        return false
+    }
+    
+    func messagesEnumerator(forRoom roomId: String) -> MXEventsEnumerator {
+        return MXEventsEnumeratorOnArray(messages: [])
+    }
+    
+    func messagesEnumerator(forRoom roomId: String, withTypeIn types: [Any]?) -> MXEventsEnumerator {
+        return MXEventsEnumeratorOnArray(messages: [])
+    }
+    
+    func relations(forEvent eventId: String, inRoom roomId: String, relationType: String) -> [MXEvent] {
+        return []
+    }
+    
+    func store(_ user: MXUser) {
+    }
+    
+    func users() -> [MXUser]? {
+        return nil
+    }
+    
+    func user(withUserId userId: String) -> MXUser? {
```

Comment:
We use the result of this function at `MXBackgroundSyncService.swift:132`. Why did you remove fetching our user from the store?

Reply:
Thanks for the catch. I missed it. I implemented `MXBackgroundSyncService.swift:132` in another way.

Reply:
Cool, thanks